apps/server/src/modules/h5p-library-management/service/h5p-library-management.service.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-console */
+import {
+	H5PConfig,
+	cacheImplementations,
+	LibraryManager,
+	ContentTypeCache,
+	IUser,
+	LibraryAdministration,
+} from '@lumieducation/h5p-server';
+import ContentManager from '@lumieducation/h5p-server/build/src/ContentManager';
+import ContentTypeInformationRepository from '@lumieducation/h5p-server/build/src/ContentTypeInformationRepository';
+import { Injectable } from '@nestjs/common';
+import { ContentStorage, LibraryStorage } from '@src/modules/h5p-editor/service';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { readFileSync } from 'fs';
+import { parse } from 'yaml';
+
+const h5pConfig = new H5PConfig(undefined, {
+	baseUrl: '/api/v3/h5p-editor',
+	contentUserStateSaveInterval: false,
+	setFinishedEnabled: false,
+});
+
+@Injectable()
+export class H5PLibraryManagementService {
+	contentTypeCache: ContentTypeCache;
+
+	contentTypeRepo: ContentTypeInformationRepository;
+
+	libraryManager: LibraryManager;
+
+	libraryAdministration: LibraryAdministration;
+
+	libraryWishList: string[];
+
+	constructor(private readonly libraryStorage: LibraryStorage, private readonly contentStorage: ContentStorage) {
+		const kvCache = new cacheImplementations.CachedKeyValueStorage('kvcache');
+		this.contentTypeCache = new ContentTypeCache(h5pConfig, kvCache);
+		this.libraryManager = new LibraryManager(
+			this.libraryStorage,
+			undefined,
+			undefined,
+			undefined,
+			undefined,
+			undefined,
+			h5pConfig
+		);
+		this.contentTypeRepo = new ContentTypeInformationRepository(this.contentTypeCache, this.libraryManager, h5pConfig);
+		const contentManager = new ContentManager(this.contentStorage);
+		this.libraryAdministration = new LibraryAdministration(this.libraryManager, contentManager);
+		const librariesYamlContent = readFileSync(Configuration.get('H5P_EDITOR__LIBRARY_LIST_PATH') as string, {
```

Comment:
Using of Configuration.get inside injectables is not allowed. Please use the nest configuration module and service with a self written config interface for this module. You find examples in code. Check starting the server module file or config file.

Reply:
solved: 8045a528dfd709bd0669ec3f560d20bb6c6a6a5f

Reply:
Search "  from '@nestjs/config';  "

Reply:
solved: 7be444624772b6560c45b7ed6224e6d861a3ae4c